sdk/src/Core/Amazon.Runtime/Credentials/DefaultAWSCredentialsIdentityResolver.cs

```diff
@@ -53,38 +53,69 @@ public DefaultAWSCredentialsIdentityResolver()
             };
         }
 
+        #region Helper methods for migration from FallbackCredentialsFactory
+
         /// <summary>
         /// Search the environment for configured AWS credentials. The search includes
         /// environment variables, "default" AWS credentials profiles and EC2 instance metadata.
         /// </summary>
+        /// <param name="clientConfig">
+        /// Optional config object that can be used to specify a different profile programatically (via the <see cref="Profile"/> property).
+        /// </param>
         /// <returns>AWSCredentials that can be used when creating AWS service clients.</returns>
-        public static AWSCredentials GetCredentials()
+        public static AWSCredentials GetCredentials(IClientConfig clientConfig = null)
         {
-            return _defaultInstance.Value.ResolveIdentity();
+            return _defaultInstance.Value.ResolveIdentity(clientConfig);
         }
 
         /// <summary>
         /// Search the environment for configured AWS credentials. The search includes
         /// environment variables, "default" AWS credentials profiles and EC2 instance metadata.
         /// </summary>
+        /// <param name="clientConfig">
+        /// Optional config object that can be used to specify a different profile programatically (via the <see cref="Profile"/> property).
+        /// </param>
         /// <returns>AWSCredentials that can be used when creating AWS service clients.</returns>
-        public static Task<AWSCredentials> GetCredentialsAsync()
+        public static Task<AWSCredentials> GetCredentialsAsync(IClientConfig clientConfig = null)
         {
-            return _defaultInstance.Value.ResolveIdentityAsync();
+            return _defaultInstance.Value.ResolveIdentityAsync(clientConfig);
         }
 
-        BaseIdentity IIdentityResolver.ResolveIdentity()
+        #endregion
+
+        BaseIdentity IIdentityResolver.ResolveIdentity(IClientConfig clientConfig) 
+            => ResolveIdentity(clientConfig);
+
+        public AWSCredentials ResolveIdentity(IClientConfig clientConfig)
         {
-            return ResolveIdentity();
+            var profile = clientConfig?.Profile;
+            if (profile != null)
+            {
+                var source = new CredentialProfileStoreChain(profile.Location);
+                if (source.TryGetProfile(profile.Name, out CredentialProfile storedProfile))
+                {
+                    return storedProfile.GetAWSCredentials(source, true);
+                }
+
+                throw new AmazonClientException($"Unable to find the \"{profile.Name}\" profile specified in the client configuration.");
+            }
+
+            return InternalGetCredentials();
         }
 
+        async Task<BaseIdentity> IIdentityResolver.ResolveIdentityAsync(IClientConfig clientConfig, CancellationToken cancellationToken) => 
```

Comment:
Use `Task.FromResult` to avoid having to do the `Task.Run` which is costlier.
```csharp
Task<BaseIdentity> IIdentityResolver.ResolveIdentityAsync(IClientConfig clientConfig, CancellationToken cancellationToken) =>
    Task.FromResult<BaseIdentity>(ResolveIdentity(clientConfig));

public Task<AWSCredentials> ResolveIdentityAsync(IClientConfig clientConfig, CancellationToken cancellationToken = default) =>
    Task.FromResult(ResolveIdentity(clientConfig));
```